File: Chapter 04/C#/Section_4_2_Downward_Dependency/Before/PresentationLayer.cs
namespace Chapter04.Section_4_2.Before
{
    // A fake UI layer class to illustrate the bad dependency
    public class PresentationLayer
    {
        public static PresentationLayer Instance { get; } = new PresentationLayer();
        public void UpdateStatusLabel(string text)
        {
            Console.WriteLine($"[UI UPDATE]: {text}");
        }
    }
}

File: Chapter 04/Python/section_4_2_downward_dependency/before/presentation_layer.py
"""
A fake UI layer class to illustrate the bad dependency.
"""
class PresentationLayer:
    _instance = None

    @classmethod
    def get_instance(cls):
        if cls._instance is None:
            cls._instance = PresentationLayer()
        return cls._instance

    def update_status_label(self, text: str):
        print(f"[UI UPDATE]: {text}")